config/config/src/index.ts

```diff
@@ -379,6 +380,19 @@ export async function getConfig (opts: {
       }
     }
   }
+
+  const supportedArchitecturesKeys = ['os', 'cpu', 'libc'] as const satisfies Array<keyof SupportedArchitectures>
+  for (const key of supportedArchitecturesKeys) {
+    const values = pnpmConfig[key]
+    if (values != null) {
+      pnpmConfig.supportedArchitectures ??= {}
+      pnpmConfig.supportedArchitectures[key] = [
+        ...pnpmConfig.supportedArchitectures[key] ?? [],
+        ...typeof values === 'string' ? [values] : values,
+      ]
+    }
+  }
```

Comment:
@zkochan This block of code would merge CLI's `supportedArchitectures` with `rootManifest.pnpm.supportedArchitectures`. But I'm not sure if this is a desired behavior. If you want one to completely override the other, let me know.

Reply:
Probably the CLI values should override the ones in the pnpmConfig.

Reply:
https://github.com/pnpm/pnpm/pull/9745/commits/e36c8c0390a31e03f1d26a4baf4973d328d3a6ac